File: problems/rice/Prob10.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package problems.rice;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;
import problems.Input;

/**
 *
 * @author rootie
 */
public class Prob10 {

    public static void main(String[] args) throws FileNotFoundException {
        // get file obj
        Scanner scanner = new Scanner(new File(Input.JPATH + "Prob10.in.txt"));
        String scoreLengo[] = {"love", "15", "30", "40"};
        int pl[] = new int[2];
        boolean deuce = false;
        while (scanner.hasNextLine()) {
            // do logic on game
            if (pl[0] + pl[1] == 0) {
                System.out.println("Game start");
            }
            pl[(Integer.parseInt(scanner.nextLine()) - 1)]++;
            if ((pl[0] > 3 || pl[1] > 3) || pl[0] == 3 && pl[1] == 3) {
                switch (Math.abs(pl[0] - pl[1])) {
                    case 0:
                        System.out.println("deuce");
                        deuce = true;
                        break;
                    case 1:
                        if (deuce) {
                            System.out.println((pl[0] > pl[1]) ? "Advantage Player 1" : "Advantage Player 2");
                        } else {
                            System.out.println((pl[0] > pl[1]) ? "Game Player 1" : "Game Player 2");
                            pl = new int[2];
                        }
                        break;
                    case 2:
                        System.out.println((pl[0] > pl[1]) ? "Game Player 1" : "Game Player 2");
                        deuce = false;
                        pl = new int[2];
                        break;
                    default:
                        System.out.println((pl[0] > pl[1]) ? "Game Player 1" : "Game Player 2");
                        pl = new int[2];
                        break;
                }
                continue;
            }
            System.out.println((pl[0] == pl[1]) ? scoreLengo[pl[0]] + "-all" : scoreLengo[pl[0]] + "-" + scoreLengo[pl[1]]);
        }
        scanner.close();
    }

}

File: problems/rice_py/prob10.py
import path
import copy

pl = [-1, -1]

lengo = {
    0 : "love",
    1 : "15",
    2 : "30",
    3 : "40"
}

def p_score():
    if pl[0] == pl[1]:
        if pl[0] < 3:
            print(lengo[pl[0]] + "-all")
        else:
            print("deuce")
    elif pl[0] > 3 or pl[1] > 3:
        if abs(pl[0] - pl[1]) == 2:
            do_win()
        elif pl[0] == pl[1]:
            print("deuce")
        elif pl[0] > pl[1]:
            print("Advantage Player 1")
        else:
            print("Advantage Player 2")
    else:
        print(lengo[pl[0]] + "-" + lengo[pl[1]])

def do_win():
    if pl[0] > pl[1]:
        print("Game Player 1")
    else:
        print("Game Player 2")
    for i in range(0, 2):
        pl[i] = -1

path = path.j_input + "Prob10.in.txt"
file = open(path, "r")

while True:
    try:
        pl[int(file.readline()) - 1] += 1
        if pl[0] == -1 or pl[1] == -1:
            pl[0] += 1
            pl[1] += 1
            print("Game start")
    except ValueError:
        break
    if (pl[0] == 4 or pl[1] == 4) and (pl[0] + pl[1]) < 7:
        do_win()
        continue
    p_score()